invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -373,12 +374,45 @@ def get_locations(self, obj):
             if geometry:
                 geo_type = geometry["type"]
                 # PIDS-FIXME: Scalable enough?
-                # PIDS-FIXME: Implement Box and Polygon serialization
                 if geo_type == "Point":
                     serialized_location["geoLocationPoint"] = {
-                        "pointLatitude": geometry["coordinates"][0],
-                        "pointLongitude": geometry["coordinates"][1],
+                        "pointLongitude": str(geometry["coordinates"][0]),
+                        "pointLatitude": str(geometry["coordinates"][1]),
                     }
+                if geo_type == "Polygon":
+                    # geojson has a layer of nesting before actual coordinates
+                    coords = geometry["coordinates"][0]
+                    # First we see if we have a box
+                    box = False
+                    if len(coords) in [4, 5]:
+                        # A box polygon may wrap around with 5 coordinates
+                        x_coords = set()
+                        y_coords = set()
+                        for coord in coords:
+                            x_coords.add(coord[0])
+                            y_coords.add(coord[1])
+                        if len(x_coords) == 2 and len(y_coords) == 2:
+                            x_coords = sorted(x_coords)
+                            y_coords = sorted(y_coords)
+                            serialized_location["geoLocationBox"] = {
+                                "westBoundLongitude": str(x_coords[0]),
+                                "eastBoundLongitude": str(x_coords[1]),
+                                "southBoundLatitude": str(y_coords[0]),
+                                "northBoundLatitude": str(y_coords[1]),
+                            }
+                            box = True
```

Comment:
If I understand correctly, a titled box is not considered a box e.g.

```
       +
     /   \
   /       \
 /           \
+ 90deg angle +
 \           / 
   \       / 
     \   /   
       +
```

```
coords = [(0,0), (2,2), (3,-3), (5, -1)]
```

Is this problematic? Probably not, right? I guess `geoLocationBox` is just a more succinct way of encoding 4 points (not sure there is tremendous savings there though...).

Reply:
The box for DataCite is defined by just two points (lower left and upper right), so a tilted box won't count as a box. It's fine for this serializer though, because a tilted box would be serialized as a polygon.